### PR TITLE
implement options dialog for renv

### PR DESCRIPTION
**Pull request by kevinushey**
This PR makes it possible to configure a subset of `renv`'s options and settings through the RStudio UI.

Screen Shot 2020-02-07 at 10 29 53 AM

Some background: `renv` has its own notion of project settings (tied to a particular project), and user configuration (tied to a particular user). I'll discuss those both briefly so the PR is a bit easier to review.

## Project Settings

Project settings for `renv` are persisted as part of the project, in a file called `renv/settings.dcf`. These project settings are accessed and modified through the `renv::settings` helper API; e.g. the cache might be enabled with:

```
renv::settings$use.cache(TRUE)
```

Because these settings are persisted in the project and understood directly by `renv`, all RStudio needs to do when a user sets these options in the UI is emit the appropriate console commands to change those project settings. If curious, there is more documentation in `?renv::settings`.

## User Configuration

User configuration is more ad-hoc. It is done through R options -- for example:

```
options(renv.config.auto.snapshot = TRUE)
```

would instruct `renv` to automatically snapshot and update the lockfile after installing or removing a package. More documentation is available in `?renv::config`.

Currently, because `renv` does not maintain its own single source of truth as to where these user settings should be stored, I've instead opted to have these stored as project preferences. It's worth discussing if this is the correct decision.

This PR does not yet add support for Python within `renv`; that will be considered as part of a follow-up PR after we're sure we have the right framework for managing these.

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/projects/ui/prefs/ProjectPreferencesDialog.java [hunk 1]
```diff
@@ -153,16 +157,74 @@ public void onError(ServerError error)
...
    
-   private void emitRenvConsoleActions(RProjectRenvOptions options)
+   private void emitRenvConsoleActions(RProjectRenvOptions newOptions)
    {
-      if (options.useRenv == renvOptions_.useRenv)
+      EventBus events = pEventBus_.get();
+      
+      // renv is off and staying off -- nothing to do
+      if (!newOptions.useRenv && !renvOptions_.useRenv)
+      {
```
Comment (adamconroy) [inline src/gwt/src/org/rstudio/studio/client/projects/ui/prefs/ProjectPreferencesDialog.java]: Perhaps return in this empty block? It looks like there's no code that executes if this block is hit.

File: src/gwt/src/org/rstudio/studio/client/projects/ui/prefs/ProjectPreferencesDialog.java [hunk 2]
```diff
@@ -153,16 +157,74 @@ public void onError(ServerError error)
...
+         events.fireEvent(new SendToConsoleEvent(action, true, true));
          return;
+      }
       
-      String renvAction = options.useRenv
-            ? "renv::activate()"
-            : "renv::deactivate()";
+      // turning renv on -- just call activate
+      //
+      // TODO: we need to apply project options here, but we need to call
```
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/projects/ui/prefs/ProjectPreferencesDialog.java]: We generally avoid committing TODOs to master -- can this be tracked in an issue? (Or can we describe what exactly will be broken before the work is completed?)

**Comment by slopp** [comment]
@kevinushey not sure if this is too late given the state of 1.3, but I would be really excited about two other renv options:

- Enabling and picking your Python. It can be really hard for newer users interacting with reticulate to get a correct and isolated Python environment without entering the terminal and learning virtualenv... a dialog box in the UI would go really far to helping them make the right choice, and even accomplish a simple  task like seeing what choice was made!!

- Setting your repo option. There has been some confusion with repository option precedence, esp given the global repo option. If a user does want to select a repository for a specific project, e.g., in order to "freeze" a project against a repository checkpoint in RSPM or MRAN, it can be very hard to achieve without knowing how to modify the renv settings file.

**Comment by kevinushey** [comment]
@slopp we're planning to make these changes for v1.4 rather than v1.3 -- this work is unfortunately coming in a little too hot.

**Comment by jmcphers** [comment]
@kevinushey is rethinking this experience for 1.4, so this PR is being closed in favor of that work.

**Pull request closed without merging**